main/ogn-aprs.go

```diff
@@ -0,0 +1,240 @@
+				log.Printf("APRS issue: " + scanner.Err().Error())
+			}
+			aprsExitChan <- true
+		}()
+
+		pgrmzTimer := time.NewTicker(100 * time.Millisecond)
+
+	loop:
+		for globalSettings.APRS_Enabled {
+			// log.Println(len(aprsIncomingMsgChan))
+			select {
+			case data := <-aprsIncomingMsgChan:
+				// APRS,qAS: aircraft beacon
+				// APRS,TCPIP*,qAC: ground station beacon
+
+				res := rex.FindStringSubmatch(data)
+				if res == nil { // no match
+					if strings.Contains(data, "TCPIP*") {
+						// log.Printf("GW data: " + data)
+					} else {
+						log.Printf("No match for: " + data)
+					}
+					continue
+				} else if len(res) == 0 { // no group capture
+					log.Printf("No group capture: " + data)
+				} else if len(res) > 0 && len(res[14]) > 0 {
+					// for i := range res {
+					// 	fmt.Printf("%s(%d)|", res[i], len(res[i]))
+					// }
+					// fmt.Printf("\n")
+					// fmt.Printf("%+v\n", res)
+
+					lat, err := strconv.ParseFloat(res[5][:2], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					lat_m, err := strconv.ParseFloat(res[5][2:len(res[5])-1], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					lat_m3d, err := strconv.ParseFloat(res[12][:1], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					if strings.Contains(res[5], "S") {
+						lat = -lat
+					}
+					lon, err := strconv.ParseFloat(res[6][:3], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					lon_m, err := strconv.ParseFloat(res[6][3:len(res[6])-1], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					lon_m3d, err := strconv.ParseFloat(res[12][1:], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					if strings.Contains(res[6], "W") {
+						lon = -lon
+					}
+					// log.Printf(res[6][:2] + " " + res[6][2:len(res[6])-1] + " " + res[12][1:])
+
+					track, err := strconv.ParseFloat(res[8], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					speed, err := strconv.ParseFloat(res[9], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+					alt, err := strconv.ParseFloat(res[10], 64)
+					if err != nil {
+						log.Printf(err.Error())
+					}
+
+					details, err := hex.DecodeString(res[14][:2])
+					if err != nil {
+						log.Fatal(err)
+					}
+					detail_byte := details[0]
+					addr_type := detail_byte & 0b00000011
+					acft_type := (detail_byte & 0b00111100) >> 2
+
+					msg := OgnMessage{
+						Sys:       res[1],
+						Time:      0,
+						Addr:      res[2],
+						Addr_type: int32(addr_type),
+						Acft_type: fmt.Sprintf("%d", acft_type),
+						Lat_deg:   float32(lat + lat_m/60 + lat_m3d/1000),
+						Lon_deg:   float32(lon + lon_m/60 + lon_m3d/1000),
+						Alt_msl_m: float32(alt * 0.3048),
+						Track_deg: track,
+						Speed_mps: speed * 0.514444}
+
+					fmt.Printf("%+v\n", res)
+					// fmt.Printf("%+v\n", msg)
+
+					importOgnTrafficMessage(msg, data)
+				}
+			case <-pgrmzTimer.C:
```

Comment:
Is this some copy-paste leftover?
I don't think you want to write into this channel - it is intended for communication with ogn-rx-eu.

Reply:
it's actually where the (not so) magic happens. at this stage we use the ogn.go function that feeds stratux, but we do not use the ognOutgoingMsgChan (nor ognIncomingMsgChan btw). this way we avoid a lot of code duplication (essentially all this long function)

Reply:
unless you have concerns, i see this as resolved

Reply:
Sorry, the github review functionality is a bit strange here. My concern was regaring the last line, the pgrmzTimer. Not sure why you would want to handle pgrmz here at all?

Using the importOgnTrafficMessage is fine of course.